internal/QueryProcessor.py

```diff
@@ -10,17 +55,79 @@ class QueryProcessor:
     #       -> send query to a local model (it will figure out what's important and query-worthy)
     #       -> send local model's response to google language
     def query_vision(self, query, image):
-        pass
+        query_label = predict(query)
+        if query_label == "offline_question":
+            return local_llm_predict(query)
+        elif query_label == "basic_question":
+            pass
+        elif query_label == "complex_question":
+            language_response = self.query_google_language(query)
+            system_prompt = f"""
+            You will answer my question based on the following information:
+
+            1. Relevant entity information extracted from the question (language_response): {language_response}
+
+            Please synthesize the above information and accurately respond to my query. If you are uncertain about any aspects, please indicate the reasons for your uncertainty and make reasonable inferences based on the available information.
+            """
+            response = self.query_gpt(query, system_prompt)
+            return response
+        elif query_label == "vision":
+            vision_response = self.query_google_vision(image)
```

Comment:
Please double check the specification to make sure you understand how vision queries are supposed to work. 

`query` given by the call will not be sufficient (for ex, image could be a landmark building, query could be `how tall is this`, so before querying language, we need to reform that query to `how tall is [name of building as returned by google vision]`.